src/Zenith.sol

```diff
@@ -54,21 +67,61 @@ contract Zenith is HostPassage, AccessControlDefaultAdminRules {
     /// @param admin - the address that will be the initial admin.
     constructor(address admin) AccessControlDefaultAdminRules(1 days, admin) {}
 
+    /// @notice Submit a rollup block with block data submitted via calldata.
+    /// @param header - see _submitBlock.
+    /// @param blockData - full data for the block supplied directly.
+    /// @param v - see _submitBlock.
+    /// @param r - see _submitBlock.
+    /// @param s - see _submitBlock.
+    /// @custom:reverts see _submitBlock.
+    /// @custom:emits see _submitBlock.
+    function submitBlock(BlockHeader memory header, bytes memory blockData, uint8 v, bytes32 r, bytes32 s) external {
```

Comment:
should probablt make this `bytes calldata` otherwise it will get calldatacopy'd to memory and cost significant gas in memexpansion

Reply:
yeah i knowwwww, it's annoying, the reason i had it like this is because the shared `_submitBlock` function needs to have `committedData` as `memory` because `bytes memory encodedHashes` can't be `calldata`...

Reply:
but i think i can fix this, sec

Reply:
hmm yeah, actually no. i think as long as we're abi-encoding the `blockData` into the `blockCommitment` it's going to `calldatacopy`... 🤔

Reply:
okay, we could do this instead: https://github.com/init4tech/zenith/pull/7/commits/4aaca568b202872c41b5993ded0737a6431ed71f

make Zenith agnostic to the submitted `blockData`. once we wean ourselves off emitting `blockData` (eg remove the BlockData event) the contracts will no longer calldatacopy the `blockData` :)

Reply:
per our discussion in DMs, we can also include a `bytes32 blockDataHash` in the function params which is also hashed in the Sequencer signature. Zenith still ignores the raw `blockData`, but the Node will reject the block if `blockDataHash != keccak256(blockData)`

this is reminding me: we need to think about how the block numbers of rollup blocks won't necessarily coincide with the sequence number on Zenith. because sequence numbers on Zenith can be consumed by invalid blocks.

Reply:
implemented here: https://github.com/init4tech/zenith/pull/7/commits/de9c38e3ce324a399902ee179a1ebb284d11dbec